File: yuv_convert/src/main/java/com/zmy/yuv_convert/output/PackedByteBufferOutput.java
package com.zmy.yuv_convert.output;

import com.zmy.yuv_convert.Format;

import java.nio.ByteBuffer;

public class PackedByteBufferOutput extends Output<ByteBuffer> {
    public PackedByteBufferOutput(Format format) {
        super(format);
    }

    @Override
    public ByteBuffer getOutput() {
        container.position(0);
        return container;
    }
}

File: yuv_convert/src/main/java/com/zmy/yuv_convert/output/PackedByteBufferOutput.kt
package com.zmy.yuv_convert.output

import com.zmy.yuv_convert.Format
import java.nio.ByteBuffer

class PackedByteBufferOutput(format: Format) : Output<ByteBuffer>(format) {
    override fun getOutput(): ByteBuffer {
        container!!.position(0)
        return container!!
    }
}